.github/workflows/superset-e2e.yml

```diff
@@ -109,7 +108,7 @@ jobs:
           run: cypress-run-all
       - name: Upload Artifacts
         uses: actions/upload-artifact@v4
-        if: failure()
+        if: steps.check.outputs.python || steps.check.outputs.frontend
```

Comment:
isn't this statement different than `failure()` which will catch if any of the above run steps fail?

Reply:
from my understanding `if: failure()` is similar to `if: steps.check.outcome == 'failure'`. The previous bash script would receive params, like `python` or `frontend` and `exit 1` if it discovered a file matched the pattern, then subsequent jobs used that failure as a trigger. Now `if failure()` is broader than `if: steps.check.outcome == 'failure'`, and will trigger the job if ANY of the previous step failed. It feels like this is right though